client/landing/gutenboarding/components/domain-picker/button.tsx

```diff
@@ -27,11 +28,18 @@ const DomainPickerButton: FunctionComponent = () => {
 	// Without user search, we can provide recommendations based on title + vertical
 	const { siteTitle, siteVertical } = useSelect( select => select( ONBOARD_STORE ).getState() );
 
-	let search = domainSearch.trim();
-	if ( ! search && isFilledFormValue( siteTitle ) ) {
-		search = siteTitle;
-	}
-
+	/**
+	 * Debounce our input + HTTP dependent select changes
+	 *
+	 * Rapidly changing input generates excessive HTTP requests.
+	 * It also leads to jarring UI changes.
+	 *
+	 * @see https://stackoverflow.com/a/44755058/1432801
+	 */
+	const inputDebounce = 200;
+	const [ debouncedDomainSearch ] = useDebounce( domainSearch.trim(), inputDebounce );
+	const search =
+		! debouncedDomainSearch && isFilledFormValue( siteTitle ) ? siteTitle : debouncedDomainSearch;
```

Comment:
Would it make sense to also debounce on `siteTitle`? Right now, we'd still fire a ton of network requests if the user types a site title, wouldn't we?

(Or just debounce on the entire `search` expression, I guess?)